python/ray/autoscaler/batching_node_provider.py

```diff
@@ -230,6 +242,16 @@ def terminate_node(self, node_id: str) -> Optional[Dict[str, Any]]:
                 f"{node_type}. Skipping termination request."
             )
 
-        self.scale_request.desired_num_workers[node_type] -= 1
-        self.scale_request.workers_to_delete.add(node_id)
+        # Scale down entire replica if part of a TPU podslice being deleted
+        node_replica_index = self.node_data_dict[node_id].replica_index
+        if node_replica_index:
+            for worker in self.replicas_to_nodes[node_replica_index]:
+                # Check if node has already been scheduled to delete
+                if node_id not in self.scale_request.workers_to_delete:
```

Comment:
Do we promise to add all Pods in a single multi-host TPU PodSlice that we want to scale down to `workers_to_delete` at the same time, or do we add them one by one? That is,

```python
# Assume we have a TPU PodSlice with 4 hosts.
# case 1
T=0 workers_to_delete = [Pod1, Pod2, Pod3, Pod4]

# case 2
T=0 workers_to_delete = [Pod1]
T=1 KubeRay deletes Pod1
T=2 workers_to_delete = [Pod2]
T=3 KubeRay deletes Pod2
...


```

Reply:
The current implementation follows case #1, scaling down the entire PodSlice at the same time. Is that the desired behavior?

Reply:
Case 1 is better for us.